packages/noya-state/src/reducers/themeReducer.ts

```diff
@@ -666,6 +668,48 @@ export function themeReducer(
         draft.selectedSymbolGroup = '';
       });
     }
+    case 'addGradientAsset': {
+      const [, name, value] = action;
+
+      return produce(state, (draft) => {
+        const assets = draft.sketch.document.assets;
+
+        const newAsset: Sketch.GradientAsset = {
+          _class: 'MSImmutableGradientAsset',
+          do_objectID: uuid(),
+          name: name,
+          gradient: value,
+        };
+
+        assets.gradientAssets.push(newAsset);
+        assets.gradients.push(value);
+      });
+    }
+    case 'removeGradientAsset': {
+      const [, id] = action;
+      const ids = typeof id === 'string' ? [id] : id;
+
+      return produce(state, (draft) => {
+        const assets = draft.sketch.document.assets;
+
+        assets.gradientAssets = assets.gradientAssets.filter((g, index) => {
```

Comment:
I think this loop can leave `assets.gradients` in a different state depending on its length and the ids passed, and I'm assuming it's supposed to match `assets.gradientAssets`.

Maybe we can look up the indexes to remove first, and then remove them in a separate pass

```ts
const indexes = assets.gradientAssets.flatMap(
  (g, index) => ids.includes(g.do_objectID) ? [index] : []
)

indexes.reverse().forEach((index) => {
  // splice from both arrays
})
```